app/controllers/expeditions_controller.rb

```diff
@@ -10,18 +12,13 @@ def show
 
   def new
     @expedition = Expedition.new
+    @expedition_type = ExpeditionType.find(params[:format])
   end
 
   def create
-    @expedition = current_user.expeditions.build(expedition_params)
-    @expedition.start_time = DateTime.now
-    if @expedition.save
-      flash[:notice] = 'You sent Expedition!'
-      # claim_worker
-      redirect_to expeditions_index_path(current_user.id)
-    else
-      flash[:alert] = @expedition.errors.full_messages
-    end
+    @dragon = params[:expedition][:dragon_id]
+    flash[:alert] = resolve_expedition
```

Comment:
Dużo fajnej pracy zostało zrobione w tych 3 PR, które teraz oglądam. Trudno się do czegoś przyczepić, to pierwsze miejsce, które znalazłem, gdzie chciałbym coś poprawić.

Mamy łańcuch metod, które zależą od siebie:
`create ` -> `resolve_expedition` -> `create_expedition`

`resolve_expedition` i `create_expedition` zwracają string z informacją co się wydarzyło.
Fajnie, gdybyśmy mogli informację o tym, jaki jest resultat, uzyskać z jakiejś klasy, która obsługuje cały ten mechanizm (zamiast z metody kontrolera, którego główym zadaniem jest posiadanie akcji powiązanych z URLami). Dobrze byłoby też, żeby metoda która zwraca rezultat mogła być wykonywana wielokrotnie.

Wyobrażam to sobie w taki sposób:

```ruby
def create
  dragon = Dragon.find(params[:expedition][:dragon_id])
  expedition = CreateExpedition.new(dragon)
  expedition.perform
  flash[:alert] = expedition.result_message

  redirect_to expeditions_index_path # nie podaję tu ID usera, bo domyślam się że każdy user może przeglądać tylko swoje (nie sprawdziłem tego, jeśli jest inaczej to trzeba przekazać ID)
end
```